### ISSUE TITLE
Error when syncing groups that contain external users

### ISSUE THREAD

**Issue by mcalhoun** [awslabs/ssosync#54]
**Describe the bug**
When syncing a group that contains external users, an exception is thrown

**To Reproduce**
1. Create a Google Group that has a mix of internal and external users (i.e., a slack channel)
2. Run the code
3. See error

```
START RequestId: 7178a6cd-fde1-4e96-9269-0836561644e0 Version: $LATEST
time="2021-08-11T17:49:12Z" level=info msg="Syncing AWS users and groups from Google Workspace SAML Application"
time="2021-08-11T17:49:12Z" level=info msg=syncing sync_method=groups
time="2021-08-11T17:49:12Z" level=debug msg="get google groups" query="name:AWS-*"
time="2021-08-11T17:49:12Z" level=debug msg="get google users and groups and its users"
time="2021-08-11T17:49:12Z" level=debug msg="get group members from google" group=AWS-SSO-BillingAdmin
time="2021-08-11T17:49:12Z" level=debug msg="get users" group=AWS-SSO-BillingAdmin
time="2021-08-11T17:49:12Z" level=debug msg="get user" group=AWS-SSO-BillingAdmin id=axl@gnr.com
time="2021-08-11T17:49:13Z" level=debug msg="get group members from google" group=AWS
time="2021-08-11T17:49:13Z" level=debug msg="get users" group=AWS
time="2021-08-11T17:49:13Z" level=debug msg="get user" group=AWS id=axl@gnr.com
time="2021-08-11T17:49:13Z" level=debug msg="get user" group=AWS id=slash@gnr.com
time="2021-08-11T17:49:13Z" level=debug msg="get user" group=AWS id=somrandomstring@gnr.slack.com
runtime error: index out of range [0] with length 0: boundsError
[{"path":"github.com/aws/aws-lambda-go@v1.23.0/lambda/errors.go","line":39,"label":"lambdaPanicResponse"},{"path":"github.com/aws/aws-lambda-go@v1.23.0/lambda/function.go","line":36,"label":"(*Function).Invoke.func1"},{"path":"runtime/panic.go","line":965,"label":"gopanic"},{"path":"runtime/panic.go","line":88,"label":"goPanicIndex"},{"path":"work/ssosync/ssosync/internal/sync.go","line":490,"label":"(*syncGSuite).getGoogleGroupsAndUsers"},{"path":"work/ssosync/ssosync/internal/sync.go","line":279,"label":"(*syncGSuite).SyncGroupsUsers"},{"path":"work/ssosync/ssosync/internal/sync.go","line":685,"label":"DoSync"},{"path":"work/ssosync/ssosync/cmd/root.go","line":55,"label":"glob..func1"},{"path":"github.com/spf13/cobra@v1.1.3/command.go","line":852,"label":"(*Command).execute"},{"path":"github.com/spf13/cobra@v1.1.3/command.go","line":960,"label":"(*Command).ExecuteC"},{"path":"github.com/spf13/cobra@v1.1.3/command.go","line":897,"label":"(*Command).Execute"},{"path":"reflect/value.go","line":476,"label":"Value.call"},{"path":"reflect/value.go","line":337,"label":"Value.Call"},{"path":"github.com/aws/aws-lambda-go@v1.23.0/lambda/handler.go","line":124,"label":"NewHandler.func1"},{"path":"github.com/aws/aws-lambda-go@v1.23.0/lambda/handler.go","line":24,"label":"lambdaHandler.Invoke"},{"path":"github.com/aws/aws-lambda-go@v1.23.0/lambda/function.go","line":64,"label":"(*Function).Invoke"},{"path":"reflect/value.go","line":476,"label":"Value.call"},{"path":"reflect/value.go","line":337,"label":"Value.Call"},{"path":"net/rpc/server.go","line":377,"label":"(*service).call"},{"path":"runtime/asm_amd64.s","line":1371,"label":"goexit"}]
END RequestId: 7178a6cd-fde1-4e96-9269-0836561644e0
REPORT RequestId: 7178a6cd-fde1-4e96-9269-0836561644e0	Duration: 2862.59 ms	Billed Duration: 2863 ms	Memory Size: 128 MB	Max Memory Used: 52 MB	Init Duration: 159.38 ms	
runtime error: index out of range [0] with length 0
boundsError
```
**Expected behavior**

Expected all internal Google users to be synced without an error and external users skipped

**Additional context**

Also tried using the `UserMatch` filter, but as you can see [here](https://github.com/awslabs/ssosync/blob/13800ad4f6239d13054bac2ba52aff97bf958a13/internal/sync.go#L705) it is not respected when using the default (sync_method=groups) sync method.

**Comment by ChrisPates** [comment]
Apologies for the delay in getting to this. I've got a code change i'm testing that should raise a debug level message for external users but continue syncing.

**Comment by ChrisPates** [comment]
Resolved by release v2.0.2
